src/compiler/binder.ts

```diff
@@ -657,19 +657,23 @@ namespace ts {
                 const saveExceptionTarget = currentExceptionTarget;
                 const saveActiveLabelList = activeLabelList;
                 const saveHasExplicitReturn = hasExplicitReturn;
-                const isIIFE = containerFlags & ContainerFlags.IsFunctionExpression && !hasSyntacticModifier(node, ModifierFlags.Async) &&
-                    !(node as FunctionLikeDeclaration).asteriskToken && !!getImmediatelyInvokedFunctionExpression(node);
+                const isIIFELike =

```

Comment:
```suggestion
                const isImmediatelyInvoked =
```

(or `isImmediate` for even more old-school flavor)